java/src/main/java/com/ncc/aif/ThreadedValidationEngine.java

```diff
@@ -310,14 +312,22 @@ public Resource getReport() {
             if (!ignored) {
                 List<RDFNode> focusNodes = SHACLUtil.getTargetNodes(shape.getShapeResource(), dataset);
                 smd.targetCount = focusNodes.size();
+                if (smd.targetCount > 0) {
+                    logger.debug("Collected " + smd.targetCount + " target node(s) for " +
+                            shape.getShapeResource().getLocalName() + ", d=" + (System.currentTimeMillis() - start));
+                }
 
                 List<RDFNode> filtered = focusNodeFilter != null ?
                         focusNodes.stream().filter(focusNodeFilter).collect(Collectors.toList()) :
                         focusNodes;
                 smd.filteredTargetCount = filtered.size();
+                if (smd.targetCount != smd.filteredTargetCount) {
+                    logger.debug("--> After filter, " + smd.filteredTargetCount + " node(s) remain.");
+                }
```

Comment:
Just create single log containing targetCount and filteredTargetCount

Reply:
I didn't do this because I wanted to keep the logs as simple as possible and not include additional information unless warranted (that is, don't mention filter count unless something was filtered out).  Also, I guess I wanted to focus on the collection time, and wasn't sure how long the filtering might take and whether that should be included in the duration.

How about something like this:

`logger.debug("Collected {} target node(s) {}for {}, d={}", smd.targetCount,`
`(smd.targetCount == smd.filteredTargetCount) ? "" : "(" + smd.filteredTargetCount + " after filter) ",`
`shape.getShapeResource().getLocalName(), (System.currentTimeMillis() - start));`

Sorry, couldn't get formatting quite right.  Anyway, it only writes the filtered number if relevant, and should only incur the cost of the string concat if the `targetCount != filterCount`.